### PR TITLE
refactor(meta): always persist compaction config.

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Previously, when accessing a compaction group's compaction config:
1. If it has been set explicitly (and persisted) before, it's used.
2. otherwise, a hard-coding `default config` is used.

The problem with this approach is, if the `default config` is changed later (e.g. hard-coding default value is changed), case 2 may start to use a different compaction config unconsciously. This can be risky especially for configs expected to be immutable for a group, e.g. `split_by_state_table`.

With this PR, whenever a new compaction group is created, it's guaranteed its compaction config is persisted as well. So that only case 1 is allowed.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/meta/src/hummock/manager/mod.rs [hunk 1]
```diff
@@ -2014,11 +2024,7 @@ where
         let compaction = read_lock!(self, compaction).await;
         for (group_id, status) in &compaction.compaction_statuses {
             if let Some(levels) = version.levels.get(group_id) {
-                let cg = self
-                    .compaction_group_manager
-                    .read()
-                    .await
-                    .get_compaction_group_config(*group_id);
+                let cg = configs.get(group_id).unwrap();
```
Comment (Li0k) [inline src/meta/src/hummock/manager/mod.rs:2027]: Directly `look up` could avoid the `unwrap`

File: src/meta/src/hummock/manager/mod.rs [hunk 2]
```diff
@@ -762,11 +767,15 @@ where
             .generate::<{ IdCategory::HummockCompactionTask }>()
             .await?;
 
-        let group_config = self
+        let group_config = match self
             .compaction_group_manager
             .read()
             .await
-            .get_compaction_group_config(compaction_group_id);
+            .try_get_compaction_group_config(compaction_group_id)
```
Comment (Li0k) [inline src/meta/src/hummock/manager/mod.rs:774]: If we use get_or_insert_compaction_group_config to initialize cg, then why can't we get config?

Comment (zwang28) [reply src/meta/src/hummock/manager/mod.rs:774]: When the last table of a compaction group is deleted, the compaction group (and its config) is destroyed as well. Then a compaction task for this gorup may come later and cannot find conifg.

Comment (Li0k) [reply src/meta/src/hummock/manager/mod.rs:774]: Thanks , need some comments ?

**Review by Li0k — APPROVED** [review]
LGTM, thanks for the PR

**Pull request merged**